contracts/extensions/OldRolesFactory.sol

```diff
@@ -0,0 +1,48 @@
+/*
+  This file is part of The Colony Network.
+
+  The Colony Network is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  The Colony Network is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with The Colony Network. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+pragma solidity >=0.5.3;
+pragma experimental ABIEncoderV2;
+
+import "./../ColonyDataTypes.sol";
+import "./../IColony.sol";
+import "./../ColonyAuthority.sol";
+import "./OldRoles.sol";
+
+
+contract OldRolesFactory is ColonyDataTypes {
```

Comment:
I wish we had type inference in Solidity when only one factory would have sufficed. I vaguely remember reading something about this in v0.5.0 breaking changes or a more recent release but couldn't find it this morning.

In the absence of this 👆 if we can at least have an interface which these factories to comply with (my preference would be for the actual `interface` keyword and `external` functions it mandates).